File: apps/api/src/ideaforge/schemas/agent.py
"""Agent schemas."""

from __future__ import annotations

from pydantic import BaseModel


class AgentInfo(BaseModel):
    id: str
    name: str
    category: str
    cost: str
    description: str
    limits: str
    requires_key: str | None
    enabled: bool
    has_api_key: bool


class AgentRunStatus(BaseModel):
    agent_name: str
    status: str
    duration_seconds: float | None = None
    result_summary: str | None = None
    error_message: str | None = None

File: apps/web/src/types/agent.ts
export interface AgentInfo {
  id: string;
  name: string;
  category: string;
  cost: "free" | string;
  description: string;
  limits: string;
  requires_key: string | null;
  enabled: boolean;
  has_api_key: boolean;
}

export interface AgentRunStatus {
  agent_name: string;
  status: "pending" | "running" | "completed" | "failed";
  duration_seconds: number | null;
  result_summary: string | null;
  error_message: string | null;
}
